From 760b116de97eef9386b6507113ffab7092cad0ae Mon Sep 17 00:00:00 2001
From: Haihua Hu <jared.hu@nxp.com>
Date: Tue, 19 Apr 2016 10:51:14 +0800
Subject: [PATCH 1/6] [glplugin] glimagesink: need to clean window_id when
 state change form READY to NULL

When application change pipeline state NULL->READY and then READY->NULL,
glimagesink will not clear glsink->window_id. After that, when application
change state NULL->READY, the new_window_id is equal to window_id, glimagesink
will not set window handle. It will use the internal window but not the window
create by application.

Upstream-Status: Accepted [https://bugzilla.gnome.org/show_bug.cgi?id=765241]


Signed-off-by: Haihua Hu <jared.hu@nxp.com>
---
 ext/gl/gstglimagesink.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/ext/gl/gstglimagesink.c b/ext/gl/gstglimagesink.c
index b627aca..241b8ed 100644
--- a/ext/gl/gstglimagesink.c
+++ b/ext/gl/gstglimagesink.c
@@ -1057,6 +1057,8 @@ gst_glimage_sink_change_state (GstElement * element, GstStateChange transition)
         gst_object_unref (glimage_sink->context);
         glimage_sink->context = NULL;
       }
+      
+      glimage_sink->window_id = 0;
 
       if (glimage_sink->other_context) {
         gst_object_unref (glimage_sink->other_context);
-- 
1.9.1

